src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -16,154 +20,182 @@ namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
-
-        private async void Tracks_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
-        {
-            if (PART_ArtistColumn == null)
-                return;
+                    AttachEvents(newItem);
+                    return newItem;
+                }));
+            }
 
-            if (Collection is AlbumViewModel)
+            if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Remove)
             {
-                foreach (var track in Collection.Tracks)
+                foreach (var item in e.OldItems)
                 {
-                    var artists = await track.GetArtistItemsAsync(track.TotalArtistItemsCount, 0).ToListAsync();
-
-                    if (artists.Count > 1)
+                    var target = TrackItems.FirstOrDefault(x => x.Track == item);
+                    if (target is not null)
                     {
-                        PART_ArtistColumn.Visibility = Visibility.Visible;
-                        return;
+                        _trackItems.Remove(target);
+                        DetachEvents(target);
                     }
                 }
-                PART_ArtistColumn.Visibility = Visibility.Collapsed;
             }
-            else
+
+            if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Move)
             {
-                PART_ArtistColumn.Visibility = Visibility.Collapsed;
+                for (var i = 0; i < e.OldItems.Count; i++)
+                    _trackItems.Move(i + e.OldStartingIndex, i + e.NewStartingIndex);
             }
+
+            if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Reset)
+            {
+                foreach (var item in TrackItems)
+                    DetachEvents(item);
+
+                _trackItems.Clear();
+            }
+
+            if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Replace)
+                throw new NotImplementedException();
         }
 
-        /// <inheritdoc/>
-        protected override async Task LoadMore()
+        private void ZuneTrackCollection_Loaded(object sender, RoutedEventArgs e)
         {
-            if (Collection == null)
-                return;
+            Loaded -= ZuneTrackCollection_Loaded;
+        }
 
-            if (!Collection.PopulateMoreTracksCommand.IsRunning)
-                await Collection.PopulateMoreTracksCommand.ExecuteAsync(25);
+        private void ZuneTrackCollection_Unloaded(object sender, RoutedEventArgs e)
+        {
+            Unloaded -= ZuneTrackCollection_Unloaded;
+
+            foreach (var item in TrackItems)
+                DetachEvents(item);
+        }
+
+        private void Item_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(ZuneTrackCollectionItem.ShouldShowArtistList))
```

Comment:
This seems hacky :)

Reply:
The only alternative is to create a new event, which is overkill here.